### ISSUE TITLE
ROUND built-in function

### ISSUE THREAD

**Issue by cuongdo** [MaterializeInc/materialize#1327]
We should support the [ROUND()](https://www.postgresql.org/docs/current/functions-math.html#id-1.5.8.8.7.2.2.18.1.1) function for rounding floating point and decimal values to the specified precision.

Here's a motivating example query:

```sql
CREATE VIEW dataflow_records_percent
AS
SELECT name, CAST(records AS float) / total_records * 100.0 AS percent
FROM mz_records_per_dataflow_global, (
	SELECT SUM(records) total_records FROM mz_records_per_dataflow_global
)
```

This generates:
```sql
| name                                        | percent               |
|---------------------------------------------+-----------------------|
| Dataflow: q07                               | 22.81698018272876     |
| Dataflow: q17                               | 14.022501292639964    |
| Dataflow: q03                               | 10.966007089067482    |
| Dataflow: q02                               | 8.988949688367274     |
| Dataflow: q19                               | 8.750740338262625     |
```

I'd love to be able to round the `percent` column to 2 digits after the decimal point.

**Comment by lmatz** [comment]
Hi @cuongdo! I am interested in implementing it, but I have some questions:
1. In [ROUND()](https://www.postgresql.org/docs/current/functions-math.html#id-1.5.8.8.7.2.2.18.1.1), the first `ROUND()` function accepts 1 argument while the second one requires an additional number `s` for specifying the number of digits after the decimal point. 
It seems that the [output_type](https://github.com/MaterializeInc/materialize/blob/master/src/expr/scalar/func.rs#L1663) of all the functions is deduced before execution, then how does the return type of `ROUND()`, e.g. a `decimal`, determine its `scale` if `s` is not a literal but comes from some subquery whose result can only be known at runtime? 
2. If we directly use the scale of the input `decimal` as the scale of the output `decimal` in the output type to bypass the above question, then the behavior of the `ROUND()` would be:<br/>
`SELECT ROUND(1.567, 2)
=>
1.570`
instead of 
`SELECT ROUND(1.567, 2)
=>
1.57`<br/>
`SELECT ROUND(1.567, 4)
=>
1.567`
instead of
`SELECT ROUND(1.567, 4)
=>
1.5670`<br/>
Is this what you expect?

**Comment by rjnn** [comment]
cc @benesch who will likely have answers as he implemented the decimal support.

**Comment by benesch** [comment]
Yep, the behavior you’ve specified looks right to me. We should double check That that is what MSSQL does, though, as their decimal type works similarly to ours.

**Comment by benesch** [comment]
Yep, this is how MSSQL works:
https://docs.microsoft.com/en-us/sql/t-sql/functions/round-transact-sql?view=sql-server-ver15#remarks

Thanks for taking this on, @lmatz! Let me know if you have any other
questions.

**Comment by wangandi** [comment]
Resolved by #2110
